### ISSUE TITLE
Add proxy support for using SteamAPI

### ISSUE THREAD

**Issue by Crockan** [louislam/uptime-kuma#2381]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

api.steampowered.com is not quite available in some complex network environments, is it possible to apply the proxy option to SteamAPI traffic to avoid problems such as DNS pollution?

### ✔️ Solution

Add an option so that third-party website traffic such as api.steampowered.com can be routed through a proxy.

**Comment by rezzorix** [comment]
Good point; maybe you can get the one who made the [PR](https://github.com/louislam/uptime-kuma/pull/492)  for this monitor to add it?

Alternatively you could also call the steamapi via https (which returns you json results) that you can monitor keywords. HTTP(S) Keyword monitor supports proxies… Further reading in my comments here: https://github.com/louislam/uptime-kuma/issues/872

**Comment by Crockan** [comment]
> 

hmmm, that's an interesting method, many thanks!
